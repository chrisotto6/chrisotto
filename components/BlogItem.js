import Image from './Image'
import Link from 'next/link'
import Tag from '@/components/Tag'
import formatDate from '@/lib/utils/formatDate'

const BlogItem = ({ slug, date, title, summary, tags, image, readingTime }) => (
  <article
    // prettier-ignore
    className="display:block w-full gap-6 space-y-2 rounded-xl xl:grid xl:grid-cols-2 xl:items-start dark:hover:bg-sky-700/40 hover:bg-gray-100"
    data-testid="list-blog-item"
  >
    <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
      <Image className="w-full rounded-xl" alt={title} src={image} width={450} height={250} />
    </Link>
    <div className="space-y-6">
      <div className="space-y-5 xl:col-span-3">
        <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
          <div>
            <h2 className="text-2xl font-bold leading-8 tracking-tight text-gray-900 dark:text-gray-100">
              {title}
            </h2>
            <dl>
              <dt className="sr-only">Published on</dt>
              <dd className="inline-block text-base font-medium leading-6 text-gray-500 dark:text-gray-400">
                <time dateTime={date}>{formatDate(date)}</time>
                <p>{readingTime.text}</p>
              </dd>
            </dl>
          </div>
          <div className="prose max-w-none text-gray-500 dark:text-gray-400">{summary}</div>
        </Link>
        <div className="flex flex-wrap" data-testid="tag-container">
          {tags.map((tag) => (
            <Tag key={tag} text={tag} />
          ))}
        </div>
      </div>
    </div>
  </article>
)

export default BlogItem
