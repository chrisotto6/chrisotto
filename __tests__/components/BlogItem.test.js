import { render } from '@testing-library/react'
import BlogItem from '@/components/BlogItem'

it('renders blogItem correctly', () => {
  const { container } = render(
    <BlogItem
      slug="slug"
      date="2021-02-19"
      title="test"
      summary="blah blah blah"
      tags={['test', 'another']}
      image="/test"
      readingTime={{ text: '2 min read', minutes: 1.61, time: 96600, words: 322 }}
    />
  )
  expect(container).toMatchSnapshot()
})
