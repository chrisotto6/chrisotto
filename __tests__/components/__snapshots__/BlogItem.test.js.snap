// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blogItem correctly 1`] = `
<div>
  <article
    class="display:block w-full gap-6 space-y-2 rounded-xl xl:grid xl:grid-cols-2 xl:items-start dark:hover:bg-sky-700/40 hover:bg-gray-100"
    data-testid="list-blog-item"
  >
    <a
      class="text-gray-900 dark:text-gray-100"
      href="/blog/slug"
    >
      <img
        alt="test"
        class="w-full rounded-xl"
        data-nimg="1"
        decoding="async"
        height="250"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Ftest&w=3840&q=75"
        srcset="/_next/image?url=%2Ftest&w=640&q=75 640w, /_next/image?url=%2Ftest&w=750&q=75 750w, /_next/image?url=%2Ftest&w=828&q=75 828w, /_next/image?url=%2Ftest&w=1080&q=75 1080w, /_next/image?url=%2Ftest&w=1200&q=75 1200w, /_next/image?url=%2Ftest&w=1920&q=75 1920w, /_next/image?url=%2Ftest&w=2048&q=75 2048w, /_next/image?url=%2Ftest&w=3840&q=75 3840w"
        style="color: transparent; max-width: 100%; height: auto;"
        width="450"
      />
    </a>
    <div
      class="space-y-6"
    >
      <div
        class="space-y-5 xl:col-span-3"
      >
        <a
          class="text-gray-900 dark:text-gray-100"
          href="/blog/slug"
        >
          <div>
            <h2
              class="text-2xl font-bold leading-8 tracking-tight text-gray-900 dark:text-gray-100"
            >
              test
            </h2>
            <dl>
              <dt
                class="sr-only"
              >
                Published on
              </dt>
              <dd
                class="inline-block text-base font-medium leading-6 text-gray-500 dark:text-gray-400"
              >
                <time
                  datetime="2021-02-19"
                >
                  February 19, 2021
                </time>
                <p>
                  2 min read
                </p>
              </dd>
            </dl>
          </div>
          <div
            class="prose max-w-none text-gray-500 dark:text-gray-400"
          >
            blah blah blah
          </div>
        </a>
        <div
          class="flex flex-wrap"
          data-testid="tag-container"
        >
          <a
            class="mr-3 text-sm font-medium uppercase text-primary-500 hover:text-primary-600 dark:hover:text-primary-400"
            href="/tags/test"
          >
            test
          </a>
          <a
            class="mr-3 text-sm font-medium uppercase text-primary-500 hover:text-primary-600 dark:hover:text-primary-400"
            href="/tags/another"
          >
            another
          </a>
        </div>
      </div>
    </div>
  </article>
</div>
`;
